ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -124,42 +113,120 @@
+                              (get-in original-subscription [:extra-fields :subscriber-id]))
+        subscriber-id (or original-subscriber
+                          (:SubscriberId (json/decode (:metadata new-subscription) true)))]
+    (common-ingest-checks request-context provider-id)
+    (check-subscription-management-acls request-context subscriber-id provider-id)
+    (perform-subscription-ingest request-context new-subscription headers)))
 
 (defn delete-subscription
   "Deletes the subscription with the given provider id and native id."
   [provider-id native-id request]
   (let [{:keys [body content-type headers request-context]} request
-        _ (common-ingest-checks request-context provider-id)
-        concept-type :subscription
-        concept (first (mdb/find-concepts request-context
-                                          {:provider-id provider-id
-                                           :native-id native-id
-                                           :exclude-metadata false
-                                           :latest true}
-                                          concept-type))]
-    (check-subscription-ingest-permission request-context concept provider-id)
-    (let [concept-attribs (-> {:provider-id provider-id
-                               :native-id native-id
-                               :concept-type concept-type}
-                              (api-core/set-revision-id headers)
-                              (api-core/set-user-id request-context headers))]
-      (info (format "Deleting %s %s from client %s"
-                    (name concept-type) (pr-str concept-attribs) (:client-id request-context)))
-      (api-core/generate-ingest-response headers
-                                         (api-core/format-and-contextualize-warnings
-                                          (ingest/delete-concept
-                                           request-context
-                                           concept-attribs))))))
+        subscriber-id (when-let [subscription (first (mdb/find-concepts
+                                                      request-context
+                                                      {:provider-id provider-id
+                                                       :native-id native-id
+                                                       :exclude-metadata false
+                                                       :latest true}
+                                                      :subscription))]
+                        (get-in subscription [:extra-fields :subscriber-id]))
+        concept-attribs (-> {:provider-id provider-id
+                             :native-id native-id
+                             :concept-type :subscription}
+                            (api-core/set-revision-id headers)
+                            (api-core/set-user-id request-context headers))]
+
+    (common-ingest-checks request-context provider-id)
```

Comment:
We want this check to be above line 209, so you don't have to make the http call to metadata-db when provider is invalid. The original code is good. You just need to delete this line.